apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/DatabaseManagerImpl.java

```diff
@@ -42,10 +47,35 @@
     private DbProperties baseDbProperties; // main database properties
     private PropertiesHolder propertiesHolder;
     private TransactionalDataSource currentTransactionalDataSource; // main/shard database
-    private Map<Long, TransactionalDataSource> connectedShardDataSourceMap = new ConcurrentHashMap<>(); // secondary shards
+
+    private CacheLoader<Long, TransactionalDataSource> loader = new CacheLoader<>() {
+        public TransactionalDataSource load(Long shardId) throws CacheLoader.InvalidCacheLoadException {
+            log.debug("Put DS shardId = '{}' into cache...", shardId);
+            TransactionalDataSource dataSource = createAndAddShard(shardId);
+            if(dataSource == null){
+                throw new CacheLoader.InvalidCacheLoadException("Value can't be null");
+            }
+            return dataSource;
+        }
+    };
+
+    private RemovalListener<Long, TransactionalDataSource> listener = dataSource -> {
+        if (dataSource.wasEvicted()) {
+            String cause = dataSource.getCause().name();
+            log.debug("Evicted DS, shutdown shardId = '{}', cause = {}", dataSource.getKey(), cause);
+            dataSource.getValue().shutdown();
+        }
+    };
+
+    private LoadingCache<Long, TransactionalDataSource> connectedShardDataSourceMap = CacheBuilder.newBuilder()
+            .maximumSize(6)
+            .expireAfterAccess(15, TimeUnit.MINUTES)
+            .weakKeys()
+            .removalListener(listener)
+            .build(loader);
+
```

Comment:
Please use constants explicitly declared in class

Reply:
I guess It's a good idea to define thees constants in the property file. It gives us an easy way to change the cache configuration.

Reply:
Added constants